### PR TITLE
Feature/zen 28163

### PR DESCRIPTION
**Pull request by morr-zenoss**
ZEN-28163: fix v2 query service to capture and report bad RPN expression, rather than blowing up.
ZEN-26776: same fix for v1 query service.
Refactored (with help from @jplouis) v1 code as part of ZEN-26776 fix - got rid of stream-based code (which wasn't really working as stream-based).
Also, fixed annoying log messages in OpenTSDBMetricStorage.java - bumped levels down to make them more appropriate to the conditions.

### CODE REVIEW

File: central-query/src/main/java/org/zenoss/app/metricservice/api/impl/JacksonResultsWriter.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public void writeResults(Writer writer, List<MetricSpecification> queries, Bucke
             writer.write(resultJson);
     }
 
-    private SeriesQueryResult makeResults(List<MetricSpecification> queries, Buckets<IHasShortcut> buckets,
+    public SeriesQueryResult makeResults(List<MetricSpecification> queries, Buckets<IHasShortcut> buckets,
```
Comment (jplouis) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/impl/JacksonResultsWriter.java]: Can the public "writeResults" above be removed?

File: central-query/src/main/java/org/zenoss/app/metricservice/api/impl/OpenTSDBMetricStorage.java [hunk 2]
```diff
@@ -450,14 +450,14 @@ private static String parseAggregation(String v) {
...
-            log.warn("Bad downsample or multiplier. Returning original downsample value of {}.", downsample);
+            log.debug("Bad downsample or multiplier. Returning original downsample value of {}.", downsample);
             return downsample;
         }
         long duration = Utils.parseDuration(downsample);
         String aggregation = parseAggregation(downsample);
         long newDuration = (long) (duration / downsampleMultiplier);
         if (newDuration <= 0) {
-            log.warn("Applying value {} of downsampleMultiplier to downsample value of {} would result in a request with resolution finer than 1 sec. returning 1 second.", downsampleMultiplier, downsample);
+            log.info("Applying value {} of downsampleMultiplier to downsample value of {} would result in a request with resolution finer than 1 sec. returning 1 second.", downsampleMultiplier, downsample);
```
Comment (jplouis) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/impl/OpenTSDBMetricStorage.java]: Were you seeing this log message?

File: central-query/src/main/java/org/zenoss/app/metricservice/api/metric/impl/MetricService.java [hunk 3]
```diff
@@ -141,31 +132,35 @@ private static Response makeCORS(Response.ResponseBuilder responseBuilder, Strin
...
+        return worker.run();
+    }
+
+    @Override
+    public Response query(Optional<String> id, Optional<String> start, Optional<String> end,
+                          Optional<ReturnSet> returnset, Optional<Boolean> series, Optional<String> downsample,
+                          double downsampleMultiplier, Optional<Map<String, List<String>>> tags,
+                          List<MetricSpecification> metrics) {
+        SeriesQueryResult queryResult = executeQuery(id, start, end, returnset, series, downsample, downsampleMultiplier, tags, metrics);
+        return makeCORS(Response.ok().entity(queryResult), MediaType.APPLICATION_JSON);
```
Comment (jplouis) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/metric/impl/MetricService.java]: Does passing an object to entity() automatically json serialize it?

File: central-query/src/main/java/org/zenoss/app/metricservice/api/impl/JacksonResultsWriter.java (hunk 1 above)
Comment (morr-zenoss) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/impl/JacksonResultsWriter.java]: No, it's referenced from MetricService.java (line 333).

File: central-query/src/main/java/org/zenoss/app/metricservice/api/impl/OpenTSDBMetricStorage.java (hunk 2 above)
Comment (morr-zenoss) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/impl/OpenTSDBMetricStorage.java]: I don't believe so. However, I thought that warn might be a little bit extreme for something that we're defaulting.

File: central-query/src/main/java/org/zenoss/app/metricservice/api/metric/impl/MetricService.java (hunk 3 above)
Comment (morr-zenoss) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/metric/impl/MetricService.java]: If I read correctly, passing it to entity() sets the object as the response entity. I assume the json serialization comes later in the process, as a result of setting the ResponseBuilder.type field to MediaType.APPLICATION_JSON in MakeCORS).

File: central-query/src/main/java/org/zenoss/app/metricservice/api/impl/JacksonResultsWriter.java (hunk 1 above)
Comment (jplouis) [inline central-query/src/main/java/org/zenoss/app/metricservice/api/impl/JacksonResultsWriter.java]: Not "makeResults", "writeResults"

**Pull request merged**